.github/workflows/pr-slack-notification.yaml

```diff
@@ -0,0 +1,304 @@
+      SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Fetch recent cache
+        run: |
+          gh extension install actions/gh-actions-cache
+
+          # Fetching list of cache keys........
+          cacheKeys=$(gh actions-cache list -R $REPO -B $BRANCH -L 100 | cut -f 1 )
+          
+          # Extract the most recent cache key
+          mostRecentCacheKey=$(echo "$cacheKeys" | head -n 1)
+          
+          if [ -n "$mostRecentCacheKey" ]; then
+            echo "CACHE_KEY=$mostRecentCacheKey" >> $GITHUB_ENV
+          else
+            echo "No recent cache key found, generating a new one...."
+            echo "CACHE_KEY=${{ runner.os }}-pr-cache-${{ github.run_number }}" >> $GITHUB_ENV
+          fi
+          
+          echo "Done"
+          # 'secrets.GITHUB_TOKEN' is automatically provided by GitHub Actions for each workflow run. You don't need to manually create or manage this token.
+
+      - name: Restore cache
+        id: cache-restore
+        uses: actions/cache@v2
+        with:
+          path: cache
+          key: ${{ env.CACHE_KEY }}
+
+      - name: Ensure cache directory exists and check if cache restored
+        run: |
+          mkdir -p cache
+          if [ -f cache/notified_prs.json ]; then
+            echo "Cache restored successfully."
+            cat cache/notified_prs.json
+          else
+            echo "Cache not restored or file does not exist.."
+          fi
+
+      - name: Fetch all opened pull request details using condition
+        id: fetch_all_pull_requests_using_condition
+        run: |
+          pr_infos=$(curl -s -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" \
+          "${{ env.API_URL }}?state=open&sort=created&direction=desc&per_page=100")
+          
+          # Load previous PR data if exists
+          if [ -f cache/notified_prs.json ]; then
+            previous_prs=$(cat cache/notified_prs.json)
+          else
+            previous_prs="[]"
+          fi
+
+          pr_list=""
+          new_notified_prs="[]"
+          notify=false
+
+          for pr_info in $(echo "$pr_infos" | jq -r '.[] | @base64'); do
+            _jq() {
+              echo "$pr_info" | base64 --decode | jq -r "${1}"
+            }
+
+            pr_number=$(_jq '.number')
+            pr_title=$(_jq '.title')
+            pr_user=$(_jq '.user.login')
+            pr_url=$(_jq '.html_url')
+            pr_draft=$(_jq '.draft')
+            pr_created_at=$(_jq '.created_at')
+            pr_updated_at=$(_jq '.updated_at')
+
+            pr_data=$(jq -n --arg number "$pr_number" --arg updated_at "$pr_updated_at" '{number: $number, updated_at: $updated_at}')
+            new_notified_prs=$(echo "$new_notified_prs" | jq --argjson pr_data "$pr_data" '. += [$pr_data]')
+
+            echo "New Notified PR List------ $new_notified_prs"
```

Comment:
Please remove unwanted echo statements.

Reply:
If we comment out this echo statement, it will be difficult to understand the workflow logs in the future. We won't be able to debug this process. The only way to analyze the workflow is by using echo statements. If we remove them now, we might need to modify the code later when checking the process (including necessary changes like secrets).We may want to rerun the workflow which will goes repeated unwanted noice in slack channel. This could disrupt the current working code. This makes it challenging for other developers, especially those who have not worked on this previously. Clear workflow logs help easily identify issues.

Reply:
@vaisakhkannan  Yes, logs help us understand the results at each step, but adding more logs can affect the performance especially inside the iterative statements like for/while loops and it will make. Also it will reduce the readability . For example, the screenshot you have attached shows logs for each PR detail with "Checking PR is new or not," followed by printing the list again with "New Notified PR List."

Instead, you could simply log the list of PRs retrieved by the API and the list of new or updated PRs, right?

Reply:
FYI for the scenario of this workflow, performance is not very important. Readability is highly important, though.

Reply:
@aparnamichael, @TrevCraw , I modified the code to list currently Opened PRs and List of PRs that need to be sent to Slack after that current run

<img width="971" alt="Screenshot 2024-06-27 at 4 57 56 PM" src="https://github.com/OpenLiberty/liberty-tools-intellij/assets/133869023/60b4127c-2715-4b86-9a7a-c78b9592322f">